contracts/access/Ownable.sol

```diff
@@ -62,6 +59,14 @@ abstract contract Ownable is Context {
      */
     function transferOwnership(address newOwner) public virtual onlyOwner {
         require(newOwner != address(0), "Ownable: new owner is the zero address");
+        _transferOwnership(newOwner);
+    }
+
+    /**
+     * @dev Internal function that forces an ownership change. Can be used to
+     * implement custom ownership management logic in childs contracts.
+     */
+    function _transferOwnership(address newOwner) internal virtual {
```

Comment:
We try to avoid situations where a single character in the function name changes the semantics of the function in a significant way. In this case it changes whether the `msg.sender` is required to be the owner so it would be better to use another name.

Maybe `_resetOwner` or `_resetOwnership`, what do you think?

Reply:
the `_transferOwnership` used to exist, but I get your remark. 
What about `_setOwner` ?

Reply:
Yes it used to exist, what I mentioned is a newer guideline we came up with. :smiley: 

`_setOwner` sounds fine, though I wonder if this sets a precedent for defining internal setters...

I think I want to put this PR on hold, finish up 3.4 and come back to this afterwards. I don't actually remember people asking for an internal owner setter, except perhaps the Claimable thing that you mentioned. Honestly in my opinion `Ownable` is so trivial that I think `Claimable` should be its own contract instead of derived from `Ownable`.

Reply:
That would indeed set a precedent for internal setters.

Having a `Claimable` extension of `Ownable` would make sens, except this extensions would not be able to set `_owner`. It would therefore have to duplicate code from `Ownable`. I think discoverability would suffer from this.

While I'm ok to discuss that after 3.4, I think having this feature in the 3.x branch would improve usability of this branch for users not willing to move to 0.8.0.